### PR TITLE
FISH-7 Adds more MP Metrics Metadata as MC Annotation for Query Pages (Virtual Pages)

### PR DESCRIPTION
**Pull request by jbee**
Payara changes for https://github.com/payara/monitoring-console/pull/18, details can be found there.

In this PR the main goal is to provide two additional attributes in the permanent annotation:

* `BaseUnit`: The unit considered base unit for MP Metrics
* `ScaleToBaseUnit`: Factor to convert the metric unit to its base unit as used by MP Metrics

In order to share the code responsible for unit scale factor is extracted to a utility class.

### CODE REVIEW

**Comment by jbee** [comment]
jenkins test please

File: appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/MetricUnitsUtils.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+import org.eclipse.microprofile.metrics.Metadata;
+import org.eclipse.microprofile.metrics.MetricUnits;
+
+/**
+ * Utilities around {@link MetricUnits}.
+ *
+ * @author Jan Bernitt
+ */
+public class MetricUnitsUtils {
```
Comment (rdebusscher) [inline appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/MetricUnitsUtils.java:50]: Suggestion: Use final class and private default constructor to have align with utility type usage?

File: appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/MetricUnitsUtils.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+    public static Number scaleToBaseUnit(Number value, Metadata metadata) {
+        if (!metadata.getUnit().isPresent()) {
+            return value;
+        }
+        String unit = metadata.getUnit().get();
+        return scaleToBaseUnit(value, unit);
+    }
+
+    public static Number scaleToBaseUnit(Number value, String unit) {
```
Comment (rdebusscher) [inline appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/MetricUnitsUtils.java:87]: Suggestion: Unit Test

**Comment by jbee** [comment]
@rdebusscher added tests and made class into a utility class.

**Comment by jbee** [comment]
jenkins test please

**Pull request merged**